pkg/apis/core/validation/validation_patch.go

```diff
@@ -44,11 +44,27 @@ var (
 )
 
 func openShiftValidateSecretUpdateIsTypeMutationAllowed(newSecret, oldSecret *core.Secret) bool {
-	// we allow "SecretTypeTLS" -> "kubernetes.io/tls" only
-	if oldSecret.Type == "SecretTypeTLS" && newSecret.Type == core.SecretTypeTLS {
+	// initially, this check was stricter.
+	// however, due to the platform's long history (spanning several years)
+	// and the complexity of ensuring that resources were consistently created with only one type,
+	// it is now permissible for ("", SecretTypeTLS, core.SecretTypeOpaque) type to transition to "kubernetes.io/tls".
+	//
+	// additionally, it should be noted that default values might also be applied in some cases.
+	// (https://github.com/openshift/kubernetes/blob/258f1d5fb6491ba65fd8201c827e179432430627/pkg/apis/core/v1/defaults.go#L280-L284)
+	if isOldSecretTypeMutationAllowed(oldSecret) && newSecret.Type == core.SecretTypeTLS {
 		if _, ok := whitelist[oldSecret.Namespace]; ok {
 			return true
 		}
 	}
 	return false
 }
+
+func isOldSecretTypeMutationAllowed(oldSecret *core.Secret) bool {
+	// "", core.SecretTypeOpaque seems safe because
+	//   https://github.com/kubernetes/kubernetes/blob/8628c3c4da6746b1dc967cc520b189a04ebd78d1/pkg/apis/core/validation/validation.go#L6393
```

Comment:
We should never see `""` in an the old secret of an update, right?

Reply:
Just playing it safe here, considering that these secrets could be created and then never changed, I don't think there is a way to ensure that resources were created only with one type.

Reply:
But if a client serializes a Secret that omits the type field or encodes it as `""`, its type would have been defaulted to Opaque. How would someone persist a Secret with type ""?

Reply:
isn't the default value applied on the server-side?

Reply:
Exactly. How would a Secret with type "" be persisted?

Reply:
Due to an incorrect patch applied to our server :) ?
Would you be okay to allow `core.SecretTypeOpaque` and `SecretTypeTLS` ?

Reply:
OK, thanks for your feedback, pushed, PTAL.